pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -558,6 +566,20 @@ func (pl *DynamicResources) Filter(ctx context.Context, cs fwk.CycleState, pod *
 		if nodeSelector := state.informationsForClaim[index].availableOnNodes; nodeSelector != nil && !nodeSelector.Match(node) {
 			logger.V(5).Info("allocation's node selector does not match", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclaim", klog.KObj(claim))
 			unavailableClaims = append(unavailableClaims, index)
+			continue
+		}
+
+		// If the claim is not ready to bind, cannot use this allocation.
+		if claim.Status.Allocation == nil || !pl.enableDeviceBindingConditions || !pl.enableDeviceStatus {
+			continue
+		}
+
+		bound, err := pl.isClaimReadyForBinding(claim)
```

Comment:
Maybe rename `bound` to something else:
```suggestion
		ready, err := pl.isClaimReadyForBinding(claim)
```